### DISCUSSION TITLE
Push by workflow does not trigger another workflow

### DISCUSSION THREAD

**Discussion by big-andy-coates** [community/community#38186] [Actions] [answered]
I've spent a more than a full day trying to get one workflow to push a git tag and have that trigger another workflow, without success.  Help! :)

I know from the [Github Docs](https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#triggering-a-workflow-from-a-workflow) that I need to use a PAT when pushing the changes, as the default `GITHUB_TOKEN` will cause the push to be ignored.

I know from similar issues (e.g. https://github.com/orgs/community/discussions/37103), I should set the PAT in the `actions/checkout` step using `token`, so that all subsequent git actions use it.

But still... no joy!

None of the docs seem to cover what access / permissions the PAT needs, so I'm assuming very little.  However, even if I give the PAT admin on everything... no joy.

I've a trimmed down example repo here:  https://github.com/big-andy-coates/tigger-workflow

The PAT is a classic token (not the fine-grained beta)

Where am I going wrong?

Any and all help from you lovely people very much appreciated.

Thanks

Andy

**Comment by antonblr** [comment]
If you want to match all tags you should use `'**'` in your workflow [here](https://github.com/big-andy-coates/tigger-workflow/blob/main/.github/workflows/tag-listener.yml#L5).

[Filter pattern cheat sheet](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet)

**Comment by airtower-luna** [comment, marked as answer]
You need to give the token to `actions/checkout` as an *input*, not an environment variable. Note the `with` instead of [`env` in your file](https://github.com/big-andy-coates/tigger-workflow/blob/e211a3057d944b29880aca7af1827c80879b72df/.github/workflows/tag-publisher.yml#L26):

```yaml
      - uses: actions/checkout@v3.1.0
        with:
          token: ${{ secrets.TRIGGER_GITHUB_TOKEN }}
```

**Reply by big-andy-coates** [reply]
Aha!  I knew it would be something simple.  A cut & paste error... of course.   Much obliged!!!
